### PR TITLE
Remove useless logging line

### PR DESCRIPTION
**Pull request by Jorricks**
Currently more than 50% of our scheduler logs are in the form of `Failed to get task ... Marking it as removed`.
There are two of these logging lines in this file, one already has this if else statement, the other one was missing it.
This should considerably bring down logging messages for airflow instances where the `tasks` change a lot while using the same dag.

### CODE REVIEW

**Comment by Jorricks** [comment]
@eladkal is this still on time for Airflow 2.4.0?
It would be awesome to include this in a `2.3.x` release as well.

**Review by potiuk — APPROVED** [review]

**Comment by Jorricks** [comment]
That was quick, even for you @potiuk :p

**Comment by potiuk** [comment]
For 2.4.0 for sure - we have not yet cut the branch for 2.4. I will mark it for 2.3.4 in case we have it, but it's not very likely I think ( we will go straight to 2.4.0 with all bugfixes most likely - but it's up to @jedcunningham and @ephraimbuddy to decided.).

File: airflow/models/dagrun.py [hunk 1]
```diff
@@ -728,9 +728,12 @@ def _filter_tis_and_exclude_removed(dag: "DAG", tis: List[TI]) -> Iterable[TI]:
                 try:
                     ti.task = dag.get_task(ti.task_id)
                 except TaskNotFound:
-                    self.log.error("Failed to get task for ti %s. Marking it as removed.", ti)
-                    ti.state = State.REMOVED
-                    session.flush()
+                    if ti.state == State.REMOVED:
+                        pass  # ti has already been removed, just ignore it
+                    else:
```
Comment (jedcunningham) [inline airflow/models/dagrun.py]:
```suggestion
                    if ti.state != State.REMOVED:
```

nit: we can probably just go with this.

**Review by jedcunningham — APPROVED** [review]

File: airflow/models/dagrun.py (hunk 1 above)
Comment (Jorricks) [reply airflow/models/dagrun.py]: Yes that seems fair. I think that is also explicit enough indeed.

**Pull request merged**